### PR TITLE
Training Documentation

### PR DESCRIPTION
**Pull request by baijumeswani**
This pull request adds documentation for the C, C++ training api.

Documentation currently staged at: https://baijumeswani.github.io/onnxruntime/docs/api/c/index.html

### CODE REVIEW

**Review by RyanUnderhill — APPROVED** [review]

File: docs/c_cxx/Doxyfile [hunk 1]
```diff
@@ -2251,7 +2253,7 @@ SEARCH_INCLUDES        = YES
 # preprocessor.
 # This tag requires that the tag SEARCH_INCLUDES is set to YES.
 
-INCLUDE_PATH           =
+INCLUDE_PATH           = "../../include/onnxruntime/core/session/"
```
[the remark this answers is not in the document]
Comment (baijumeswani) [reply docs/c_cxx/Doxyfile:2262]: Added a comment explaining this change.

File: include/onnxruntime/core/session/onnxruntime_c_api.h [hunk 2]
```diff
@@ -3,8 +3,15 @@
 
 // See docs\c_cxx\README.md on generating the Doxygen documentation from this file
 
-/** \mainpage C & C++ APIs
+/** \mainpage OnnxRuntime
```
[the remark this answers is not in the document]
Comment (baijumeswani) [reply include/onnxruntime/core/session/onnxruntime_c_api.h]: Updated. Please refresh the staged website to review changes.

**Review by askhade — APPROVED** [review]

**Review by natke — APPROVED** [review]

**Review by jchen351 — APPROVED** [review]
Approved for ES

**Comment by baijumeswani** [comment]
Thanks for the review. 😄

**Pull request merged**